src/Command/UnmakeEntityCommand.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Hexis\UnmakeBundle\Command;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Question\Question;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+
+class UnmakeEntityCommand extends Command
+{
+    protected static $defaultName = 'unmake:entity';
+    protected static $defaultDescription = 'Delete entities based on provided filters.';
+    private EntityManagerInterface $entityManager;
+    private ParameterBagInterface $parameterBag;
+
+    public function __construct(EntityManagerInterface $entityManager, ParameterBagInterface $parameterBag)
+    {
+        parent::__construct();
+        $this->entityManager = $entityManager;
+        $this->parameterBag = $parameterBag;
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(self::$defaultDescription);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $entityList = $this->getEntityNames();
+        $entitySelected = $this->selectEntity($input, $output, $entityList);
+
+        if (!$entitySelected) {
+            $output->writeln('<error>No valid entity selected.</error>');
+            return Command::FAILURE;
+        }
+
+        $qb = $this->entityManager->createQueryBuilder();
+        $qb->select('a')->from($entitySelected, 'a');
+
+        $filterCounter = 0;
+        do {
+            $filterCounter = $this->addFilter($input, $output, $qb, $filterCounter);
+        } while ($this->askForAnotherProperty($input, $output));
+
+
+        $result = $qb->getQuery()->getResult();
+        $output->writeln('<info>Number of rows affected:</info> ' . count($result));
+
+        if ($this->confirmDeletion($input, $output, $result)) {
+            $deletedRows = $this->deleteRows($qb, $entitySelected);
+
+            $output->writeln('<info>Number of rows deleted:</info> ' . $deletedRows);
+        } else {
+            $output->writeln('<info>Deletion canceled or no rows to delete.</info>');
+        }
+
+        return Command::SUCCESS;
+    }
+
+    private function getEntityNames(): array
+    {
+        $entityList = $this->entityManager->getMetadataFactory()->getAllMetadata();
+        return array_map(static function ($metadata) {
+            return $metadata->getName();
+        }, $entityList);
+    }
+
+    private function selectEntity(InputInterface $input, OutputInterface $output, array $entityNames): ?string
+    {
+        $helper = $this->getHelper('question');
+        $question = new Question('<question>What entity do you want to delete?</question> ');
```

Comment:
Why do we need to write `<question>`?

Reply:
When we write `<question>`, it uses different color coding in console output, which is designed to make it easier to guide the user through commands.

Reply:
I improved the code by utilizing `SymfonyStyle` class, which automatically formats the question style. 
Previously, the implementation was without `SymfonyStyle` class.